### ISSUE TITLE
Entity type:Person are being created before actually saving an organization's entity  

### ISSUE THREAD

**Issue by cyberandy** [insideout10/wordlift-plugin#265]
I've noticed, while woking with entities type:Organization, that even before saving the information on the founder an entity type:Person was created in the vocabulary. 

This should not happen.

**Comment by mcolacino** [comment]
Which version of WL? ... Please update production WL 3.4 (which merges WL 3.3.5) ... It could be a problem related to entity auto save that is now disabled for entities.

**Comment by cyberandy** [comment]
3.4 - it occurred on the blog.insideout.io/en website.

**Comment by ziodave** [comment]
@cyberandy can you describe how to reproduce it step-by-step?

In fact I tried, but I don't see the founder metabox unless I actually save (a draft of) the organization.

**Comment by ziodave** [comment]
In order to reproduce it, open test.wordlift.it, create a new entity, e.g. Microsoft, save a draft: the founder is added automatically (named 'Communications'?) and published.

**Comment by cyberandy** [comment]
![image](https://cloud.githubusercontent.com/assets/837037/18093972/0b8e9246-6ed2-11e6-9072-7e719dc5cf98.png)

here is what happens - we save the _new_ founder BEFORE the organization is saved.

**Comment by ziodave** [comment]
The _Communications_ entity already exists in the vocabulary: http://test.wordlift.it/wp-admin/post.php?post=4058&action=edit

Removing this entity seems to solve.

**Comment by ziodave** [comment]
I've dumped some debug data:

```
[ default entity identifier :: ]
[ entity ID :: 4058 ]
[ label :: Communications ]
[ value :: 4058 ]
```

The issue seems to be in `html_input` of `WL_Metabox_Field_uri.php`, which thinks an empty `$default_entity_identifier` is a URI and then tries to get an entity post:

`$entity = Wordlift_Entity_Service::get_instance()->get_entity_post_by_uri( $default_entity_identifier );`

which eventually returns the first entity post (in our test *Communications*) which is then suggested as founder.

**Comment by ziodave** [comment]
The fix takes two approaches:

1. in `Wordlift_Entity_Service->get_entity_post_by_uri` we check if the `$uri` is empty, in which case we return `null` right away,
2. in `WL_Metabox_Field_uri->html_input` we check if `$default_entity_identifier` is empty, in which case we set `$entity` to `null` and we don't even bother calling `Wordlift_Entity_Service->get_entity_post_by_uri`.

**Comment by ziodave** [comment]
merged in master

**Comment by ziodave** [comment]
Published with 3.5.4
